src/pbrt/textures.cpp

```diff
@@ -989,15 +989,23 @@ WrinkledTexture *WrinkledTexture::Create(const Transform &renderFromTexture,
 
 #if defined(PBRT_BUILD_GPU_RENDERER)
 
+// mipmaps were not fully supported in HIP
+// we use just the base resolution for now
+#ifdef __HIPCC__
+using TextureArray = cudaArray_t;
+#else
+using TextureArray = cudaMipmappedArray_t;
```

Comment:
Again as far as breaking things into smaller steps: could we have a separate PR that just rewrites the existing CUDA functionality to be in terms of the new `TextureArray` type, without any of the __HIPCC__ additions, then have the HIPCC changes subsequently come in and generalize things?